### PR TITLE
Use github.com/gofrs/flock to lock handler

### PR DESCRIPTION
**Pull request by qinqon**
**Is this a BUG FIX or a FEATURE ?**:

> Uncomment only one, leave it on its own line:
>
/kind bug
Closes https://github.com/nmstate/kubernetes-nmstate/issues/729
> /kind enhancement

**What this PR does / why we need it**:
The nmstate-handler has a lock that make it wait if another handler is
already running in the system, but the library used for that does not
work well with containers. This change replace that library with
different one that uses golang syscall.Flock behind the scine [1] and
that works fine with containers sharing node volume.

[1] github.com/nightlyone/lockfile.

**Special notes for your reviewer**:

**Release note**:

```release-note
Fix handler locking using github.com/gofrs/flock.
```

### CODE REVIEW

**Comment by qinqon** [comment]
/hold
wip

File: main.go [hunk 1]
```diff
@@ -186,6 +188,25 @@ func main() {
...
+			os.Exit(1)
+			setupLog.Error(err, "failed checking nmstatectl health")
+		}
+
+		// Handler runs with host networking so opening ports is problematic
+		// they will collide with node ports so to ensure that we reach this
+		// point (we have the handler lock and nmstatectl show is working) a
+		// file is touched and the file is checked at readinessProbe field.
+		healthyFile := "/tmp/healthy"
+		setupLog.Info("Marking handler as healthy touching healthy file", "healthyFile", healthyFile)
```
Comment (rhrazdil) [inline main.go:204]: This message sounds odd, we don't mark handler healthy here, do we? That's what the probe will do once it tries to cat that file

File: test/e2e/operator/nmstate_install_test.go [hunk 2]
```diff
@@ -85,24 +74,55 @@ var _ = Describe("NMState operator", func() {
...
+		)
+		BeforeEach(func() {
+			installNMState(defaultNMState)
+			daemonset.GetEventually(handlerKey).Should(daemonset.BeReady())
+			installOperator(operatorNamespace)
+		})
+		AfterEach(func() {
+			uninstallNMState(defaultNMState)
+			uninstallOperator(operatorNamespace)
+			installOperator("nmstate")
```
Comment (rhrazdil) [inline test/e2e/operator/nmstate_install_test.go:89]: Why is the operator installed in "nmstate" namespace at the end?
If it's done to test that operator can be installed and uninstalled at one namespace and then installed in another, wouldn't it be better to have another spec for it?

File: main.go (hunk 1 above)
Comment (qinqon) [reply main.go:204]: well we mark the container by touching the file

Comment (rhrazdil) [reply main.go:204]: Ahh, I understand now

File: test/e2e/operator/nmstate_install_test.go (hunk 2 above)
Comment (qinqon) [reply test/e2e/operator/nmstate_install_test.go:89]: The test-e2e-operator test suite spec the operator to be installed, but the test we are adding here remove the nmstate operator to check that the handler installed from different namespace is blocked until the "nmstate" one is delete. 

The install here ensure that after that test the nmstate operator is there as expected at the other test cases.

**Comment by rhrazdil** [comment]
Thank you
/approve
/lgtm

**Pull request merged**